homeassistant/components/ecowitt/sensor.py

```diff
@@ -273,8 +343,36 @@ def __init__(
         """Initialize the sensor."""
         super().__init__(sensor)
         self.entity_description = description
+        self.restored_value: StateType | datetime | date | Decimal = None
+        self.restored_last_reported: datetime | None = None
+
+    async def async_added_to_hass(self) -> None:
+        """Call when entity about to be added to hass."""
+        await super().async_added_to_hass()
+        if self._attr_native_value is not None:
+            return
+        if (sensor_last_state := await self.async_get_last_state()) is not None:
+            self.restored_last_reported = sensor_last_state.last_reported
+
+        if (sensor_data := await self.async_get_last_sensor_data()) is not None:
+            self._attr_native_value = sensor_data.native_value
+            self._attr_available = True
+            self.restored_value = sensor_data.native_value
 
     @property
-    def native_value(self) -> StateType | datetime:
+    def native_value(self) -> StateType | datetime | date | Decimal:
         """Return the state of the sensor."""
-        return self.ecowitt.value
+        if self.ecowitt.value is not None:
+            return self.ecowitt.value
+        return self.restored_value
+
+    @property
+    def available(self) -> bool:
+        """Return whether the state is based on actual reading from device."""
+        if self.ecowitt.value is not None:
+            return super().available
+        if self.restored_last_reported:
+            return (
+                self.restored_last_reported + timedelta(minutes=5)
```

Comment:
Can we create a constant for the 5 minutes and use it in the entity.py

I'm not sure if it's overengineered but we could run a task in `self.restored_last_reported + timedelta(minutes=5)`, which sets the entity to unavailable. This future will be canceled if an update comes in. So we make sure that the entity will be unavailable if no update comes in.

Reply:
That sounds reliable.

Reply:
So are you implementing it?

Reply:
I've tried it here:
https://github.com/home-assistant/core/pull/134442/commits/88f2958f7f5659983c893b94ed593ed864a965f4
But `self._attr_available` seems to make no difference in the state of the entity. Do you have an idea, why?